drivers/ieee802154/ieee802154_cc13xx_cc26xx.h

```diff
@@ -80,7 +80,7 @@ struct ieee802154_cc13xx_cc26xx_data {
 	volatile rfc_CMD_IEEE_CSMA_t cmd_ieee_csma;
 	volatile rfc_CMD_IEEE_TX_t cmd_ieee_tx;
 	volatile rfc_CMD_IEEE_RX_ACK_t cmd_ieee_rx_ack;
-	volatile rfc_CMD_RADIO_SETUP_t cmd_radio_setup;
```

Comment:
Not all chips covered by this driver actually have a PA. Of course this was a bug before but I think now it's a bug in reverse. See the solution in the cc13xx_26xx SubG driver which you could apply here as well.

Note: I'm really sorry because I'm quite sure my initial review on that type was not helpful either. :-( I didn't see that there were different types required because those lines did not appear in the diff. So you'll also have to re-introduce the #ifdef I asked you to remove, because those properties are probably missing.

Here is my original comment with some context:

> ```
> #if defined(CONFIG_SOC_CC1352P) || defined(CONFIG_SOC_CC1352P7)
> 		.pRegOverrideTxStd = NULL,
> ```
> This code seems to be redundant. As a statically initialized struct, it will be zeroed anyway. Switching to the correct structure as you've done in the .h file should be enough to fix the bug, right?

Reply:
> See the solution in the cc13xx_26xx SubG driver which you could apply here as well.

Should I also copy rf_prop_overrides_tx_std and rf_prop_overrides_tx_20 ?
Are their values also valid for 2.4G radio ?

Reply:
> Should I also copy rf_prop_overrides_tx_std and rf_prop_overrides_tx_20 ? Are their values also valid for 2.4G radio ?

No, I meant "similarly to", not copy over verbatim. But at the same time I was causing confusion because I made an error myself:

> So you'll also have to re-introduce the #ifdef I asked you to remove, because those properties are probably missing.
> ```
> #if defined(CONFIG_SOC_CC1352P) || defined(CONFIG_SOC_CC1352P7)
> 		.pRegOverrideTxStd = NULL,
> ```

You can still leave that one out. It is still initialized to NULL and therefore redundant. I have no idea what made me say that it has to be re-introduced other than I must have been really tired. Sigh.

What you have to change, though is something like this:

```diff
-volatile rfc_CMD_RADIO_SETUP_t cmd_radio_setup;
+#if defined(CONFIG_SOC_CC1352R) || defined(CONFIG_SOC_CC2652R) || defined(CONFIG_SOC_CC1352R7) || defined(CONFIG_SOC_CC2652R7)
+volatile rfc_CMD_RADIO_SETUP_t cmd_radio_setup;
+#elif defined(CONFIG_SOC_CC1352P) || defined(CONFIG_SOC_CC2652P) || defined(CONFIG_SOC_CC1352P7) || defined(CONFIG_SOC_CC2652P7)
+volatile rfc_CMD_RADIO_SETUP_PA_t cmd_radio_setup;
+#else
+BUILD_ASSERT(false, "unknown model");
+#endif /* CONFIG_SOC_CCxx52x */
```

And similarly in `ieee802154_cc13xx_cc26xx.c` around line 780:

```diff
-		.commandNo = CMD_RADIO_SETUP,
+#if defined(CONFIG_SOC_CC1352R) || defined(CONFIG_SOC_CC2652R) || defined(CONFIG_SOC_CC1352R7) || defined(CONFIG_SOC_CC2652R7)
+		.commandNo = CMD_RADIO_SETUP,
+#elif defined(CONFIG_SOC_CC1352P) || defined(CONFIG_SOC_CC2652P) || defined(CONFIG_SOC_CC1352P7) || defined(CONFIG_SOC_CC2652P7)
+		.commandNo = CMD_RADIO_SETUP_PA,
+#endif /* CONFIG_SOC_CCxx52x */
```

That is what I meant is similar to what we do in the SubG driver. But the SubG driver obviously doesn't include the 26xx models because they only have a 2.4G radio. And it uses slightly different commands.

Makes sense now?